sentry/src/main/java/io/sentry/clientreport/ClientReportRecorderImpl.java

```diff
@@ -0,0 +1,176 @@
+package io.sentry.clientreport;
+
+import io.sentry.DateUtils;
+import io.sentry.SentryEnvelope;
+import io.sentry.SentryEnvelopeItem;
+import io.sentry.SentryItemType;
+import io.sentry.SentryLevel;
+import io.sentry.SentryOptions;
+import io.sentry.transport.DataCategory;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public final class ClientReportRecorderImpl implements ClientReportRecorder {
+
+  private static volatile @Nullable ClientReportRecorderImpl SHARED_INSTANCE = null;
+
+  public static @NotNull ClientReportRecorderImpl getInstance() {
+    if (SHARED_INSTANCE == null) {
+      synchronized (ClientReportRecorderImpl.class) {
+        if (SHARED_INSTANCE == null) {
+          SHARED_INSTANCE = new ClientReportRecorderImpl();
+        }
+      }
+    }
```

Comment:
Not sure we need this double-check, the SDK is initialized in a single thread and won't access `getInstance` from multiple threads.
Its methods need to be thread-safety tho.

Reply:
I'm wondering if it needs to be a singleton or just a property in SentryOptions that don't recreate its instance, let's check how the other SDKs did it.

Reply:
[ ] discuss whether singleton is necessary

Reply:
changed to single instance on options